boa_gc/src/pointers/weak_pair.rs

```diff
@@ -0,0 +1,77 @@
+use crate::{
+    finalizer_safe,
+    internals::Ephemeron,
+    trace::{Finalize, Trace},
+    GcBox, EPHEMERON_QUEUE,
+};
+use std::cell::Cell;
+use std::ptr::NonNull;
+
+pub struct WeakPair<K: Trace + ?Sized + 'static, V: Trace + ?Sized + 'static> {
```

Comment:
In my opinion, this should be named `Ephemeron` and the current ephemeron should be named `EphemeronBox`. `WeakPair` sounds like a pair of data being managed by a `WeakGc`.